core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -606,6 +607,16 @@ private[log] class Cleaner(val id: Int,
 
       position += result.bytesRead
 
+      // Validate CRC if buffer was grown beyond max.message.bytes to read this batch.
+      // Note that we don't check every batch > max.message.bytes, just one batch is validated
+      // as a sanity check when the buffer is grown beyond max.message.bytes.
+      if (checkCrc && result.messagesRead > 0) {
+        checkCrc = false
+        val batches = records.batches.iterator
+        if (batches.hasNext && !batches.next.isValid)
+          throw new CorruptRecordException(s"Log for ${topicPartition} contains corrupt data")
```

Comment:
We should include the log segment base offset if we can in messages such as this. Also, at this point we have already allocated the batch, so is checking crc even helpful?

Reply:
I was thinking that if there is a lot of corrupt data in the logs and we managed to allocate a buffer for the first one because the size was small enough to allocate, it may be worth checking CRC to detect corruption and avoid allocating even larger buffers later.